client/landing/gutenboarding/index.tsx

```diff
@@ -58,9 +58,11 @@ window.AppBoot = async () => {
 	// Add accessible-focus listener.
 	accessibleFocus();
 
-	try {
-		checkAndRedirectIfSiteWasCreatedRecently();
-	} catch {}
+	// If site was recently created, redirect to customer site home.
+	const shouldRedirect = await checkAndRedirectIfSiteWasCreatedRecently();
+	if ( shouldRedirect ) {
+		return;
+	}
```

Comment:
For future work in another PR, this will be used to display the modal that asks user what to do. https://github.com/Automattic/wp-calypso/issues/49565#issuecomment-771585671